src/ophys_etl/modules/suite2p_registration/suite2p_utils.py

```diff
@@ -117,14 +129,31 @@ def compute_reference(frames: np.ndarray,
             frame[:] = shift_frame(frame=frame, dy=dy, dx=dx)
 
         # Create a new reference image from the highest correlated data.
-        ref_image = max_corr_frames.mean(axis=0).astype(np.int16)
+        with warnings.catch_warnings():
+            # Assuming the motion correction went well, there should be a lot
+            # of empty values in the padded area of the frames. We suppress
+            # warnings for these "Empty Slices" as they are expected.
+            warnings.filterwarnings('ignore', 'Mean of empty slice')
+            ref_image = np.nanmean(max_corr_frames, axis=0)
         # Shift reference image to position of mean shifts to remove any bulk
         # displacement.
         ref_image = shift_frame(
             frame=ref_image,
             dy=int(np.round(-max_corr_ymax.mean())),
             dx=int(np.round(-max_corr_xmax.mean()))
         )
+        # Clip the reference image back down to the original size and remove
+        # any NaNs remaining. Throw warning if a NaN is found.
+        ref_image = ref_image[pad_y:-pad_y, pad_x:-pad_x]
+        if np.any(np.isnan(ref_image)):
```

Comment:
I still don't see how it can happen because the maximum possible offset is pad_x, pad_y, and we are removing those pixels here. Have you actually encountered this situation? Can you write a test for this situation? This just seems like a code block that if I encountered it in the future, I would still be confused.

Reply:
So imagine this pathological case: 1/2 of the frames are shifted to `(max_pady, max_padx)`, and half to (-max_pady, -max_padx). The average displacement that is removed from the final reference is then (0, 0). The image would have some data in the middle of the trimmed reference frame, but the two quadrants where no data was shifted would not contain any data even after trimming. If this were to happen in real data, I would think something had happened to the scope or rig in some way. In this case you would be left with something similar to the motion boarder that shows up in the current code. In fact it might be better with this version of the code as those empty spots would be filled with a smooth value instead of wrapped data with structure which, at least in my intuition, would reduce the likelihood of chance correlation.

I can code up this specific test case to make this happen in a controlled way.